lib/ansible/modules/cloud/amazon/waf_facts.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+module: waf_facts
+short_description: Retrieve facts for WAF ACLs, Rule , Conditions and Filters.
+description:
+  - Retrieve facts for WAF ACLs, Rule , Conditions and Filters.
```

Comment:
Or aws_web_acl_facts maybe.

I haven't really come across using WAFs with ALBs - do you know what boto3 modules provide insight? I found https://aws.amazon.com/about-aws/whats-new/2016/12/AWS-WAF-now-available-on-Application-Load-Balancer/ but nothing really concrete about getting information. 

I'd think we'd want elb_application_lb_facts to provide information about which web ACL rulesets are attached, and this module to provide information about those Web ACL rules.